### DISCUSSION TITLE
Enable Heap Tracing?

### DISCUSSION THREAD

**Discussion by Bwanna** [espressif/arduino-esp32#8270] [Q&A]
Hi,
I have a rather larger Arduino project that is running, but with a heap memory issue (possible fragmentation).   

How can I **enable heap tracing** to use the additional instruction available for diagnosing this?  All I can find is using 'menuconfig', but I believe that's only available in ESP-IDF.

Goal is to measure fragmentation %.

-working with esp32 devkit

Thanks,
Steve

**Comment by mrengineer7777** [comment]
True.  You'll need to convert your project to IDF (possibly with Arduino as IDF component) to get that functionality.

**Reply by Bwanna** [reply]
That's what I thought, but nice to have it confirmed.   Something NEW to learn!  Thanks and Cheers

**Reply by HamzaHajeir** [reply]
You can build your custom arduino libraries using [esp32-arduino-lib-builder](https://github.com/espressif/esp32-arduino-lib-builder) tool, and continue by using Arduino.
